python/cudf/cudf/core/series.py

```diff
@@ -661,6 +661,115 @@ def axes(self):
+        axis : {0 or 'index'}, optional
+            Axis to truncate. Truncates the index (rows) by default.
+        copy : bool, default is True,
+            Return a copy of the truncated section.
+        Returns
+        -------
+            The truncated Series.
+        Notes
+        -----
+        If the index being truncated contains only datetime values,
+        `before` and `after` may be specified as strings instead of
+        Timestamps.
+        Examples
+        --------
+        >>> import cudf
+        >>> cs1 = cudf.Series([1, 2, 3, 4])
+        >>> cs1
+        0    1
+        1    2
+        2    3
+        3    4
+        dtype: int64
+
+        >>> cs1.truncate(before=1, after=2)
+        1    2
+        2    3
+        dtype: int64
+
+        >>> import cudf
+        >>> dates = cudf.date_range('2021-01-01', '2021-01-02', freq='s')
+        >>> cs2 = cudf.Series(range(len(dates)), index=dates)
+        >>> cs2
+        2021-01-01 00:00:00        0
+        2021-01-01 00:00:01        1
+        2021-01-01 00:00:02        2
+        2021-01-01 00:00:03        3
+        2021-01-01 00:00:04        4
+        ...
+        2021-01-01 23:59:55    86395
+        2021-01-01 23:59:56    86396
+        2021-01-01 23:59:57    86397
+        2021-01-01 23:59:58    86398
+        2021-01-01 23:59:59    86399
+        Length: 86400, dtype: int64
+
+        >>> cs2.truncate(
+        ... before="2021-01-01 23:45:18", after="2021-01-01 23:45:27")
+        2021-01-01 23:45:18    85518
+        2021-01-01 23:45:19    85519
+        2021-01-01 23:45:20    85520
+        2021-01-01 23:45:21    85521
+        2021-01-01 23:45:22    85522
+        2021-01-01 23:45:23    85523
+        2021-01-01 23:45:24    85524
+        2021-01-01 23:45:25    85525
+        2021-01-01 23:45:26    85526
+        2021-01-01 23:45:27    85527
+        dtype: int64
+
+        >>> cs3 = cudf.Series({'A':1, 'B':2, 'C':3, 'D':4})
+        >>> cs3
+        A    1
+        B    2
+        C    3
+        D    4
+        dtype: int64
+
+        >>> cs3.truncate(before='B', after='C')
+        B    2
+        C    3
+        dtype: int64
+        """
+        if axis not in (0, "index"):
+            raise ValueError(f"No axis named {axis} for object type Series")
+        ax = self.index
+
+        if not ax.is_monotonic_increasing and not ax.is_monotonic_decreasing:
+            raise ValueError("truncate requires a sorted index")
+
+        if ax is cudf.core.index.DatetimeIndex:
+            before = pd.to_datetime(before)
+            after = pd.to_datetime(after)
+
+        if before is not None and after is not None and before > after:
+            raise ValueError(f"Truncate: {after} must be after {before}")
+
+        if len(ax) > 1 and ax.is_monotonic_decreasing and ax.nunique() > 1:
+            before, after = after, before
+
+        slicer = slice(before, after)
+        result = self.loc[slicer]
+
+        if copy:
+            result = result.copy()
```

Comment:
Copy semantics should be tested explicitly.

Reply:
Missed pushing the fix, I currently do not have access to updated code.

Reply:
@VamsiTallam95 could you clarify what you don't have access to? Is this something that we can help with?

Reply:
I've addressed this. We don't actually support _not_ copying so it's mostly a doc change and validating that an error is raised.